lib/ice_nine/freezer/object.rb

```diff
@@ -16,6 +16,8 @@ class Object < self
       #
       # @return [Object]
       def self.guarded_deep_freeze(object, recursion_guard)
+        return object unless object.respond_to?(:freeze)
```

Comment:
So I had the `ThreadSafe::Cache` in mind in this case.  I feel like `undef`ing freeze is a pretty strong statement that it would not like to be frozen.  Seems like freezing the internals in this case might result in weird behavior.